### ISSUE TITLE
Users should be able to add Specialties under Abilities to their characters

### ISSUE THREAD

**Issue by jonBarcus** [jonBarcus/SoIaF_GM_Tools#24]
Right now, let's work it in the most _basic_ level.  Don't worry about XP or minimum ability requirements, etc...

**Comment by kirejm** [comment]
Working on this though I'm having trouble figuring out how editing should work; will review the docs for that and hopefully piece it together.

**Comment by kirejm** [comment]
I have everything displaying when manually adding columns to the database but I suspect that's not actually what we want here. What I'm suspecting we want is to add the ability to add an arbitrary number of specializations and also specify the number of ranks intended.

**Comment by jonBarcus** [comment]
Once we've got this part figured out...we'll be able to start getting in to the real fun stuff (adjusting attributes and specialties based on available XP).

**Comment by drumsetz** [comment]
I have a thought on this, though I'm not sure how it would work... Do we need to add a Controller for Specialty here so that we can Create a Specialty from a Character -- As Mo said, he has them displaying already -- I think we should be able to add a link for each Ability for creating a Specialty which would route to the Specialty:create.

Not entirely convinced about this though.  Thoughts?

**Comment by jonBarcus** [comment]
Yeah, not sure.  Need to look at the code again and see how things are working.  We don't want to manually add columns to the database, that's for sure.  We might need to create a new table that indicates the relationship between Ability/Specialty/Character...etc.

**Comment by drumsetz** [comment]
Been playing around with this a bit while watching March Madness... I think we already have the appropriate links in the table to tie abilities/specialties/characters.

We even have rows to link the specialty_names to ability_names; unfortunately the seeds.rb file isn't doing this properly. #25 will fix the specialty_names table so that it actually references the ability_name_id for the Ability it goes with.
